npm/private/npm_translate_lock_generate.bzl

```diff
@@ -344,13 +345,34 @@ def _gen_npm_imports(importers, packages, root_package, rctx_name, attr, registr
         # gather patches & patch args
         patches, patches_keys = _gather_values_from_matching_names(True, attr.patches, name, friendly_name, unfriendly_name)
 
+        # Apply patch from `pnpm.patchedDependencies` first
+        if pnpm_patched:
+            patch_path = "//:%s" % patched_dependencies.get(friendly_name).get("path")
```

Comment:
Can we assume the patch is in `//:`? What if it's in a `filegroup` in `//my-patches:patches`?

Reply:
Because it's a repository rule and runs before the analysis phase, the form of the label is flexible, so long as it has the correct path to the file.